app/models/dynamic_field_category.rb

```diff
@@ -10,14 +10,17 @@ class DynamicFieldCategory < ActiveRecord::Base
   validates :display_label, presence: true, uniqueness: true
   validates :metadata_form, presence: true
 
-  def as_json(_options = {})
-    {
+  def as_json(options = {})
+    json = {
       id: id,
       type: self.class.name,
       display_label: display_label,
       sort_order: sort_order,
-      children: children
+      children: children.map(&:as_json)
```

Comment:
I thought that nested as_json calls worked in such a way as to automatically call as_json on each level of discovered nested object, so you may not need to explicitly call as_json on the children here.  Might be worth a check.

Reply:
I don't think they do in the way you're talking about - I think that rubygraphql calls as_json down the line if it has a relevant type, but if you want to ensure it is called for a JSON scalar in this context you have to do it manually.

Reply:
Ah!  You're right.  I just tried it and confirmed.

Reply:
yeah, sad times.